test/helpers/kubectl.go

```diff
@@ -2916,36 +2917,64 @@ func (kub *Kubectl) GetCiliumPodOnNodeWithLabel(namespace string, label string)
 	return kub.GetCiliumPodOnNode(namespace, node)
 }
 
-func (kub *Kubectl) ciliumPreFlightCheck() error {
-	err := kub.ciliumStatusPreFlightCheck()
-	if err != nil {
-		return fmt.Errorf("status is unhealthy: %s", err)
-	}
+func (kub *Kubectl) validateCilium() error {
+	var (
+		wg     sync.WaitGroup
+		errors = make(chan error, 128)
```

Comment:
Does it need to be a buffered channel?

Reply:
Otherwise goroutines will not get to `wg.Done()` calls and we will have a deadlock (AFAIU).

Reply:
Actually, try [`golang.org/x/sync/errgroup`](https://pkg.go.dev/golang.org/x/sync/errgroup)!

Reply:
Yep, typically a good use-case for `sync/errgroup` (which is already a dep as I pulled it in recently).

Reply:
Good suggestion, did just that, thanks!

Reply:
Nice!